### PR TITLE
Luascript API feature: ParallaxLT support

### PR DESCRIPTION
**Pull request by GO-Miles**
# Hey Psych Engine github!

I'm Miles, but enough about me! I'm here to submit this **Pull Request** that adds rudimentary ParallaxLT support to Psych Engine's Luascript API.

https://github.com/ShadowMario/FNF-PsychEngine/assets/95124554/98ff057e-3436-49f1-a9d9-37dbed7c500b

<sub>early version of ParallaxLT running in a recompiled Psych Engine build</sub>

It was early 2022 when I got the idea to distort FlxSprites to mimic 3D graphics. Tons of tinkering got me an early working version that could scale and skew in a few old tech demos! Enough people kept asking about the magic under the hood that I recently decided to release a haxelib.

ParallaxSprite infographic

This was great and all for developers, but the vast majority of people who use haxeflixel are only familiar with compiled applications that have scripting support, such as Psych Engine and its Luascript API!

## The problems I wanted to solve:

- Haxeflixel's classes can make linear transformations difficult to perform and complicated to work with.
- Psych Engine's Luascript API isn't performant nor permissive enough to distort existing sprites.

## The solutions I came up with:
- I made a custom FlxSprite class that overrides draw functions to perform linear transformations based on multiple positions and scrollfactors.
- I added a ParallaxSprite map and some custom lua callbacks into Psych Engine to create, initialize, and setup members.

# **Accepting this PR** means:
- These functions will be added as lua callbacks:
  - `makeParallaxSprite(tag, image, x, y)`
  - `addParallaxSprite(tag, front)`
  - `fixateParallaxSprite(obj anchorX, anchorY, scrollOneX, scrollOneY, scrollTwoX, scrollTwoY, direct)`
- Psych Engine will ship with ParallaxLT 0.0.0 support
- The graphics will be like ***parappa but cooler***
- A little Miles icon will appear at the bottom of the special thanks! 

## This pull request was hasty, so I'm open to feedback or suggestions to improve it!

### CODE REVIEW

**Review by lunacynne — APPROVED** [review]

**Review by charlesisfeline — APPROVED** [review]

File: source/psychlua/FunkinLua.hx [hunk 1]
```diff
@@ -982,6 +983,18 @@ class FunkinLua {
 			LuaUtils.loadFrames(leSprite, image, spriteType);
 			game.modchartSprites.set(tag, leSprite);
 		});
+		Lua_helper.add_callback(lua, "makeParallaxSprite", function(tag:String, ?image:String = null, ?x:Float = 0, ?y:Float = 0) {
+			tag = tag.replace('.', '');
+			LuaUtils.resetSpriteTag(tag, true);
+			var leSprite:ParallaxSprite = new ParallaxSprite(x, y, Paths.image(image));
```
Comment (Itz-Miles) [inline source/psychlua/FunkinLua.hx:989]: @CharlesCatYT FunkinLua references the ParallaxSprites for the lua callbacks the same way ModchartSprites are, so the import is needed!

**Review by Joalor64 — APPROVED** [review]

**Comment by ShadowMario** [comment]
This is cool but I can count with my hands how many mods would actually make any kind of use for it.

**Comment by BlueColorsin** [comment]
Shadow please reconsider
having it as a tool in the first place adds for many opportunity's for modders to discover and add 
its kinda like substates, the AVERAGE psych engine coder does not use them but it is still a really useful tool for them

**Comment by cyn0x8** [comment]
ive found myself wanting to skew stage sprites based on camera pos like this a few times but ive never been able to do it
it is possible with a shader but id really rather not have to resort to that since the effect/stage composition would be ruined for ppl who have shaders disabled

**Comment by VMan-2002** [comment]
-cool addition
-tossed because of how niche it is
i wouldn't do this

**Pull request closed without merging**